### PR TITLE
GenPareto cdf and quantile functions

### PR DESCRIPTION
**Pull request by elenaehrlich**
*Description of changes:* Including cdf() and its inverse quantile() for the genpareto distribution, and added corresponding pytests in test_distribution_methods. Also changing np.inf to 10**15 in log_prob() of genpareto and gamma distributions to circumvent problems on gpu.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/gluonts/mx/distribution/genpareto.py [hunk 1]
```diff
@@ -91,6 +91,24 @@ def genpareto_log_prob(x, xi, beta):
             genpareto_log_prob(F.abs(x), xi, beta),
```
Comment (benidis) [inline src/gluonts/mx/distribution/genpareto.py:91]: Could you fix also the `-np.inf` to `-1e-15`? Same for `Gamma`. This can be a separate pr too.

Comment (elenaehrlich) [reply src/gluonts/mx/distribution/genpareto.py:91]: Thanks Konstantinos, i amended the pr to include these.

**Review by lostella — APPROVED** [review]
Looks good to me! Thanks @elenaehrlich

**Comment by benidis** [comment]
Thanks for the tests Elena. Looks good!

**Review by benidis — APPROVED** [review]
